### PR TITLE
Use the new batch endpoint in MM

### PR DESCRIPTION
**Pull request by MatanYadaev**
## Summary by CodeRabbit

- **New Features**
	- Enhanced URL validation to support a broader range of valid URLs, including localhost and specific port numbers.
	- New transformation for `websiteUrl` to ensure it is stored in lowercase.
- **Bug Fixes**
	- Reduced maximum allowable batch size for business report requests from 10,000 to 1,000.
- **Chores**
	- Updated subproject commit reference to the latest version.
	- Removed unnecessary type definitions and schemas to streamline data handling.

### CODE REVIEW

File: packages/common/src/consts/index.ts [hunk 1]
```diff
@@ -169,4 +169,4 @@ export const MatchReasonCode = {
 } as const;
 
 export const URL_PATTERN =
-  /^((https?):\/\/)?([\dA-Za-z][\w-]*\.)+[\dA-Za-z]+(\.[a-z]{2})?(\/[\w#.-]+)*(\/)?(\?[\w.-]+=[\w.-]+(&[\w.-]+=[\w.-]+)*)?(#[\w-]+)?$/;
+  /^(https?:\/\/)?((([\da-z]([\da-z-]*[\da-z])*)\.)+[a-z]{2,}|((25[0-5]|2[0-4]\d|1\d{2}|\d{1,2})\.){3}(25[0-5]|2[0-4]\d|1\d{2}|\d{1,2})|localhost)(:\d{1,5})?(\/[\w!$%&'()*+,.:;=@~-]*)*(\?([\w!$%&'()*+,.:;=@~-]+=[\w!$%&'()*+,.:;=@~-]*(&[\w!$%&'()*+,.:;=@~-]+=[\w!$%&'()*+,.:;=@~-]*)*)?)?(#[\w!$%&'()*+,.:;=@~-]*)?$/i;
```
Comment (MatanYadaev) [inline packages/common/src/consts/index.ts:172]: Synced with the regex we have in Unified API

File: services/workflows-service/src/business-report/business-report.service.ts [hunk 2]
```diff
@@ -122,8 +121,8 @@ export class BusinessReportService {
       );
     }
 
-    if (businessReportsRequests.length > 10000) {
-      throw new UnprocessableEntityException('Batch size is too large, the maximum is 10000');
+    if (businessReportsRequests.length > 1_000) {
+      throw new UnprocessableEntityException('Batch size is too large, the maximum is 1,000');
     }
```
Comment (MatanYadaev) [inline services/workflows-service/src/business-report/business-report.service.ts:126]: 1,000 is a number our system can handle.

**Review by tomer-shvadron — APPROVED** [review]

File: services/workflows-service/src/business-report/business-report.service.ts [hunk 3]
```diff
@@ -160,26 +159,18 @@ export class BusinessReportService {
...
+          customerId,
           workflowVersion,
+          withQualityControl,
+          reportType: type,
+          reports: businessWithRequests.map(({ businessReportRequest, businessId }) => ({
+            businessId,
+            websiteUrl: businessReportRequest.websiteUrl,
+            countryCode: businessReportRequest.countryCode,
+            parentCompanyName: businessReportRequest.parentCompanyName,
+            callbackUrl: `${env.APP_API_URL}/api/v1/internal/business-reports/hook?businessId=${businessId}`,
```
Comment (alonp99) [inline services/workflows-service/src/business-report/business-report.service.ts:172]: We need the callback URL?

**Review by alonp99 — APPROVED** [review]

File: services/workflows-service/src/business-report/business-report.service.ts (hunk 3 above)
Comment (liorzam) [reply services/workflows-service/src/business-report/business-report.service.ts:172]: the callback url should be related per each busniess 
so we can set this property optional and remove it from the plugin

Comment (MatanYadaev) [reply services/workflows-service/src/business-report/business-report.service.ts:172]: No, currently we don't need callback URL, the endpoint doesn't do anything.

**Pull request merged**